instrumentation/spring/spring-boot-autoconfigure-3/build.gradle.kts

```diff
@@ -13,7 +13,7 @@ otelJava {
 dependencies {
   val springBootVersion = "3.2.4"
   library("org.springframework.boot:spring-boot-starter-web:$springBootVersion")
-  compileOnly(project(":instrumentation:spring:spring-boot-autoconfigure"))
+  implementation(project(":instrumentation:spring:spring-boot-autoconfigure"))
```

Comment:
can you send this and the other non-stability change below (additional-spring-configuration-metadata.json) as a separate PR? just mark it with the milestone and I'll make sure it gets into the release

Reply:
sure - https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/11784